### PR TITLE
Task 2 (Serve SPA in AWS)

### PR DESCRIPTION
**Pull request by kastrubait**
Task: https://github.com/EPAM-JS-Competency-center/cloud-development-course-initial/blob/main/task2-serve-spa-in-s3-cloudfront/task.md
Done 16.08.2021 / deadline 16.08.2021
Score: 5

Manual:

1. bucket - http://tamiko-task2-manual.s3-website-eu-west-1.amazonaws.com
2. cloudfront - https://d3perue5xj24oa.cloudfront.net/

Auto:

1. bucket - http://tamiko-task2-auto.s3-website-eu-west-1.amazonaws.com
2. cloudfront - https://d2aifpc1btqxzw.cloudfront.net/

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "cSpell.words": ["tamiko"],
```
Comment (ragexe) [inline .vscode/settings.json:2]: 💖

File: serverless-single-page-app-plugin/index.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+  runAwsCommand(args) {
+    let command = "aws";
+    if (this.serverless.variables.service.provider.region) {
+      command = `${command} --region ${this.serverless.variables.service.provider.region}`;
+    }
+    if (this.serverless.variables.service.provider.profile) {
+      command = `${command} --profile ${this.serverless.variables.service.provider.profile}`;
+    }
+    const result = spawnSync(command, args);
```
Comment (ragexe) [inline serverless-single-page-app-plugin/index.js:40]: Suppose you might have some troubles with invalidation action. Please re-check. A possible solution is available at https://github.com/serverless/examples/issues/510

**Comment by ragexe** [comment]
@kastrubait Could you please provide a link to S3 public available bucket in accordance with the 2nd criteria? 
image

File: .vscode/settings.json (hunk 1 above)
Comment (ragexe) [reply .vscode/settings.json:2]: * for cSpell

File: serverless-single-page-app-plugin/index.js (hunk 2 above)
Comment (kastrubait) [reply serverless-single-page-app-plugin/index.js:40]: Thank you for your review. During the auto-deploy, I didn't have the described problems, but I will necessarily double-check everything.

**Review by kastrubait — COMMENTED** [review]
All links can be found in PR or ... what did you mean ??

**Comment by ragexe** [comment]
Please contact me through discord - ragexe#2978

**Comment by ragexe** [comment]
I mean that in accordance with the task you should provide the S3 bucket link available through the internet. Both of links you provided (http://tamiko-task2-manual.s3-website-eu-west-1.amazonaws.com/, http://tamiko-task2-auto.s3-website-eu-west-1.amazonaws.com/) are not available (403 error-code). Please provide me the requested link to get the highest grade. To get additional detail please contact me via discord - ragexe#2978

**Comment by ragexe** [comment]
Oh, I see that the structure should be deleted in accordance with point 2.2.2. That's O.k.
The assignment is complete. - 5 pts.

**Pull request merged**